Robot.java_03_22_17

```diff
@@ -1,518 +1 @@
-		}
-*/			
-	
-
-		if (autonomousCommand != null)
-			autonomousCommand.cancel();
-		
-		left.disable();
-		right.disable();
-		driveTrain = new RobotDrive(leftSpark, rightSpark);
-	}
-
-	/**
-	 * This function is called periodically during operator control
-	 */
-	
-	@Override
-	public void teleopPeriodic() {
-		
-		//needs a String/boolean parameter		
-		Scheduler.getInstance().run();
-		
-		//double v = s1.getDistance();
-
-		//	System.out.println(v);
-
-		System.out.println("LEFT\tgetDistance: \t\t" + leftEncoder.getDistance() + "\t\tleftRate: \t\t" + leftEncoder.getRate());
-		System.out.println("LEFT JOYSTICK" + -1*stick2.getY());
-		System.out.println("RIGHT\tgetDistance: \t\t" + rightEncoder.getDistance() + "\t\trightRate: \t\t" + rightEncoder.getRate());
-		System.out.println("RIGHT JOYSTICK" + -1*stick.getY());
-		System.out.println();
-		  if (stick2.getTrigger() && !prevTrigger) {
-			    System.out.println("Setting camera 1\n");
-			    server.setSource(cam1);
-			  }
-		  else if (!stick2.getTrigger() && prevTrigger) {
-			    System.out.println("Setting camera 0\n");
-			    server.setSource(cam0);
-			  } 
-		  prevTrigger = stick2.getTrigger();
-
-
-		
-		if (driveMode.equalsIgnoreCase("TANK")) {
-			driveTrain.tankDrive(-1*stick2.getY(), stick.getY());
-			//left, right
-		}
-		else if(driveMode.equalsIgnoreCase("ARCADE")){
-			driveTrain.arcadeDrive(-stick.getY(), -stick.getX());
-			//driveTrain.arcadeDrive(stick);
-		}
-		else if(driveMode.equalsIgnoreCase("BACK TANK")){
-			driveTrain.tankDrive(1*stick.getY(), -1*stick2.getY());
-		}
-		
-		//winch control - Halie 
-		winch.tankDrive(stick.getRawAxis(3), 0);
-		
-	}
-			
-
-	/**
-	 * This function is called periodically during test mode
-	 */
-	@Override
-	public void testPeriodic() {
-		LiveWindow.run();
-	}
-	 
-	public double feetToEncUnits(double feet) {
-		return feet*12/(6*Math.PI)*360/2400;
-	}
-	
-	
-	public void moveForward(double feet) {	
-		if (leftEncoder.getDistance() < feetToEncUnits(feet) || rightEncoder.getDistance() < feetToEncUnits(feet)) {
-			//rotations: X * 2400 / 360 
-			//feet: 
-			left.set(.5);
-			right.set(.5);
-		}
-		else {
-			left.set(0);
-			right.set(0);
-		} 
-	}
-	
-	public void moveForwardSlow(double feet) {
-		if (leftEncoder.getDistance() < feetToEncUnits(feet) || rightEncoder.getDistance() < feetToEncUnits(feet)) {
-			//rotations: X * 2400 / 360 
-			//feet: 
-			left.set(.1);
-			right.set(.1);
-		}
-		else {
-			left.set(0);
-			right.set(0);
-		} 
-	}
-	
-	
-	public boolean finishedMoving(double feet) {
-		if (leftEncoder.getDistance() < feetToEncUnits(feet) || rightEncoder.getDistance() < feetToEncUnits(feet)) {
-			return false;
-		}
-		else 
-			leftEncoder.reset();
-			rightEncoder.reset();
-			return true;
-	}
-	
-	public void changeAngle(double degrees) {
-		//want to turn until getAngle > degrees
-		//won't have to be exact 
-		
-		System.out.println(gyro.getAngle());
-		if (degrees >= 0) {
```

Comment:
The gryo you have can do absolute heading. The only things to remember is that a heading of 0 is the direction points when reset() was last called (so you can help yourself by calling it in autoInit()) and that you need to check what happens when you spin around more than once.